### PR TITLE
Added Framework Authentication (Issue #53).

### PR DESCRIPTION
**Pull request by DarinJ**
Modified `MesosScheduler.java` and `configuration.md`.  Now `mapred.mesos.framework.principal`, `mapred.mesos.framework.secretfile`, `mapred.mesos.framework.user`, and `mapred.mesos.framework.name` are configureable options.

### CODE REVIEW

**Comment by tarnfeld** [comment]
Looks awesome, thanks for the contribution! I'll give it a whirl and merge if it looks good.

File: src/main/java/org/apache/hadoop/mapred/MesosScheduler.java [hunk 1]
```diff
@@ -181,15 +183,36 @@ public synchronized void start() throws IOException {
...
-        .setName("Hadoop: (RPC port: " + jobTracker.port + ","
-                 + " WebUI port: " + jobTracker.infoPort + ")").build();
-
-      driver = new MesosSchedulerDriver(this, frameworkInfo, master);
+        .setName(conf.get("mapred.mesos.framework.name","Hadoop: (RPC port: " + jobTracker.port + ","
+                 + " WebUI port: " + jobTracker.infoPort + ")"));
+      Credential credential=null;
+
+      String frameworkPrincipal=conf.get("mapred.mesos.framework.principal");
+      if (frameworkPrincipal!=null){
```
Comment (tarnfeld) [inline src/main/java/org/apache/hadoop/mapred/MesosScheduler.java]: Would you mind just tidying up this block of code you've added to be consistent with the rest of the file (basically just some additional spaces around your `if` statements and `=` signs.

Thanks!

**Comment by tarnfeld** [comment]
Would you mind rebasing this against master? Sorry it's a pain I know!

**Comment by DarinJ** [comment]
No issues rebasing, but there's another PR in the queue.  Which order do you want to merge them?  I can probably do the rebase in the next hour, but can wait if you'd rather merge the other in first.

**Comment by tarnfeld** [comment]
The other PR is mine and isn't quite ready to go yet, so i'll sort that out. This one will definitely be merged first :smiley:

**Comment by DarinJ** [comment]
Started rebase, more merge conflicts than expected.  Shouldn't be hard but don't want to rush it before a meeting expect done by EOD tomorrow.

**Comment by DarinJ** [comment]
@tarnfeld rebased.  Tested locally to make sure the merge went fine.  Cheers.

PS: If you want a maven to enforce a checkstyle I could probably add it in the next week or so.

File: src/main/java/org/apache/hadoop/mapred/MesosScheduler.java [hunk 2]
```diff
@@ -364,10 +366,35 @@ public synchronized void start() throws IOException {
...
-      driver = new MesosSchedulerDriver(this, frameworkInfo, master);
+      FrameworkInfo frameworkInfo;
+      FrameworkInfo.Builder frameworkInfoBuilder = FrameworkInfo.newBuilder()
+              .setUser(conf.get("mapred.mesos.framework.user", "")) // Let Mesos fill in the user.
+              .setCheckpoint(conf.getBoolean("mapred.mesos.checkpoint", false))
+              .setRole(conf.get("mapred.mesos.role", "*"))
+              .setName(conf.get("mapred.mesos.framework.name", "Hadoop: (RPC port: " + jobTracker.port + ","
+                      + " WebUI port: " + jobTracker.infoPort + ")"));
+
+      Credential credential=null;
```
Comment (tarnfeld) [inline src/main/java/org/apache/hadoop/mapred/MesosScheduler.java]: Little bit of missing whitespace here

File: src/main/java/org/apache/hadoop/mapred/MesosScheduler.java [hunk 3]
```diff
@@ -364,10 +366,35 @@ public synchronized void start() throws IOException {
...
+                  .build();
+        }
+      }
+      if (credential == null) {
+        LOG.info("Creating Schedule Driver");
+        driver = new MesosSchedulerDriver(this, frameworkInfoBuilder.build(), master);
+      } else {
+        LOG.info("Creatingg Schedule Driver, attempting to authenticate with Principal: " + credential.getPrincipal()
+                + ", secret:" + credential.getSecret());
+        driver = new MesosSchedulerDriver(this, frameworkInfoBuilder.build(), master,credential);
```
Comment (tarnfeld) [inline src/main/java/org/apache/hadoop/mapred/MesosScheduler.java]: Missing a space after the comma

**Comment by tarnfeld** [comment]
Just two teeny style comments, I think we should definitely add some tools for doing checkstyle or something like that. I'm just keen to keep the code consistent across the project at the moment.

If you fix those two things i'll merge the PR right away! Thanks again.

**Comment by DarinJ** [comment]
@tarnfeld you caught me at a good time.  Updated and squashed commits.

I'll open an issue on checkstyles, I've got a few other things I'd like to contribute in the near future. Getting checkstyles will end up saving me time :).

**Pull request merged**